firmware/console/binary/tooth_logger.cpp

```diff
@@ -78,23 +81,38 @@ int copyCompositeEvents(CompositeEvent *events) {
 static void SetNextCompositeEntry(efitick_t timestamp, bool trigger1, bool trigger2,
 		bool isTDC DECLARE_ENGINE_PARAMETER_SUFFIX) {
 	uint32_t nowUs = NT2US(timestamp);
+	
 	// TS uses big endian, grumble
 	buffer[NextIdx].timestamp = SWAP_UINT32(nowUs);
 	buffer[NextIdx].priLevel = trigger1;
 	buffer[NextIdx].secLevel = trigger2;
 	buffer[NextIdx].isTDC = isTDC;
 	buffer[NextIdx].sync = engine->triggerCentral.triggerState.shaft_is_synchronized;
-	buffer[NextIdx].coil = coil;
-	buffer[NextIdx].injector = injector;
+	//ptr_buffer[NextIdx].coil = coil;
+	//ptr_buffer[NextIdx].injector = injector;
 
 	NextIdx++;
 
 	static_assert(sizeof(composite_logger_s) == COMPOSITE_PACKET_SIZE, "composite packet size");
 
-	// If we hit the end, loop
-	if (NextIdx >= sizeof(buffer) / sizeof(buffer[0])) {
+	//If we hit the end, loop
+	if ((firstBuffer) && (NextIdx >= (COMPOSITE_PACKET_COUNT/2))) {
```

Comment:
why don't you just track position in the current buffer, and when it comes time to swap them, swap the pointers? We shouldn't have to care about where we are in the larger allocation.

Reply:
Dose it make a difference? it was simpler for me this way, so i keep the old stuff and use it as much as i could, performance vise i do not think it is much difference.